src/python/WMCore/DataStructs/Run.py

```diff
@@ -142,10 +145,25 @@ def __delitem__(self, key):
     def __hash__(self):
         """
         Calculate the value of the hash
-        """
-        value = self.run.__hash__()
-        value += hash(frozenset(listitems(self.eventsPerLumi)))  # Hash that represents the dictionary
-        return value
+
+        NOTE: Python2 maxint is:
+        > python -c 'import sys; print(sys.maxint)'
+        9223372036854775807
+        so we cannot use the full range of the hexadecimal hash code because
+        it could cause an integer overflow. This is the maximum slice/value we
+        can safely use:
+        > int(15 * "f", base=16)
+        1152921504606846975
+        """
+        if isinstance(self.run, (newstr, newbytes)):
+            value = encodeUnicodeToBytesConditional(self.run, condition=PY3)
+        else:
+            value = encodeUnicodeToBytesConditional(str(self.run), condition=PY3)
+        hashValue = hashlib.sha1(value)
+        # Generate immutable sorted list of lumis
+        frozenEvents = str(sorted(listitems(self.eventsPerLumi), key=lambda x: x[0]))
+        hashValue.update(encodeUnicodeToBytes(frozenEvents))
+        return int(hashValue.hexdigest()[:15], 16)
```

Comment:
I like this! I have to say that `base=36` seemed a bit arbitrary, but having using `hexdigest()` and converting to an int in base 16 seems much more reasonable to me!

Moreover, thanks for investigating the limits of the int in py2!

Reply:
Yes, I somehow thought it would result in an smaller integer value 🤦

Reply:
(comment for future me, i know you discovered this already:) ) `int()` always give back the base 10 integer, so if you provide a string and specify it is in base 36, then it will give you back a larger integer

```
# python3
>>> x = 13245907823409857290879087
>>> int(str(x),10)
13245907823409857290879087
>>> int(str(x),36)
876957221859903199400449539217648595047
```